app/models/shipit/hook.rb

```diff
@@ -29,9 +45,14 @@ def headers
           'User-Agent' => 'Shipit Webhook',
           'Content-Type' => content_type,
           'X-Shipit-Event' => event,
+          'X-Shipit-Signature' => signature,
           'Accept' => '*/*',
         }
       end
+
+      def signature
+        DeliverySigner.new(secret).sign(payload)
```

Comment:
Good point, we should guard that and return `nil` early if secret is `nil`. Faraday will then automatically remove the header

Reply:
Since `secret` is a string column we'll expect the base case (no secret defined) to be an empty string rather than `nil`. Do we want to guard against that explicitly here (no header if empty string)? Otherwise this might be okay as-is since `secret` will always be passed as a string here.

Reply:
I'd say don't sign if the string is empty. Use `present?` or `blank?`.

Would also be worth migrating the column to allow for null values.

Reply:
Nice 🙂 I just pushed a commit, though it's using `empty?` rather than `blank?` cause I figured we don't need to worry to much about a secret value like `"   "`, fair?

Cool if we save the migration for a separate PR? I haven't thought through what `nil` secret might mean in any other scenarios, and haven't done that kind of migration before.

Reply:
> cause I figured we don't need to worry to much about a secret value like `" "`, fair?

That's not the only difference. `blank?` will work fine if `secret` is `nil`, `empty?` won't. Hence why it's preferable as it allows us to make the column nullable which is more correct.

Reply:
Ah okay, I though `nil.empty?` existed as well. Will flip it now